frontend/src/components/Nav/NavUtils.tsx

```diff
@@ -34,6 +34,11 @@ const buildCommonQueryParams = (params: GraphUrlParams): string => {
   q += `&${URLParam.DURATION}=${params.duration}`;
   q += `&${URLParam.GRAPH_OPERATION_NODES}=${params.showOperationNodes}`;
   q += `&${URLParam.REFRESH_INTERVAL}=${params.refreshInterval}`;
+
+  if (params.node && params.node.cluster) {
+    q += `&${URLParam.CLUSTERNAME}=${params.node?.cluster}`;
+  }
+
```

Comment:
I wonder if we should add it only if it's not the default/home cluster?

Reply:
Yeah, and also we should check if we are on multicluster too.

Reply:
I guess another option would be to stop using a query param and accept the fact that `cluster` is part of the node definition.  And so just change the paths to include cluster.  For example:

From:
path: '/graph/node/namespaces/:namespace/' + Paths.WORKLOADS + '/:workload',

To:
path: '/graph/node/clusters/:cluster/namespaces/:namespace/' + Paths.WORKLOADS + '/:workload',

Reply:
Maybe, but will require several changes into Kiali. For example, in list pages, today we are querying once and fetching all from all clusters, with these new approach, we should fetch and aggregate on the UI.

Reply:
I think I was not clear.  I was not suggesting we add `:cluster` everywhere, just when we are talking about a specific object, in this case the node for a node graph.  A specific node is basically defined as `cluster:namespace:name`.  And I am only talking about the UI routes at the moment.

In our current approach we are trying to manage the `cluster` portion as a query param, and the namespace and name in the path.

Reply:
Although it may be more clear, it's probably not worth the effort at this time, as the parameter approach is working.